### PR TITLE
Minor project modifications / fixes

### PR DESCRIPTION
**Pull request by NetWin**
## Changes
- Includes: https://github.com/NetWin/date-time-picker/pull/8
- Convert `@Output` to new `output()` syntax
- Convert ng-doc playground to separate demo component
- Update tsconfig, angular.json etc. to match what newer AngularCLI generates
- Fix many linter warnings
- Drop some unnecessary dependencies (e.g. `codecov` / `codelyzer`)

### CODE REVIEW

File: projects/picker/src/docs/date-time-inline/date-time-inline-demo.component.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  imports: [DatePipe, FormsModule, OwlDateTimeModule, OwlNativeDateTimeModule]
+})
+export class DateTimeInlineDemoComponent {
+  protected readonly value = signal<SingleDate | DateRange>(null);
+
+  public readonly selectMode = input<SelectMode>('range');
+
+  public readonly pickerType = input<PickerType>('both');
+
+  public readonly showTodayButton = input<boolean>(true);
```
Comment (pawcoding) [inline projects/picker/src/docs/date-time-inline/date-time-inline-demo.component.ts:26]: input signals working in NgDoc? 😮  Nice 🚀

File: projects/picker/src/lib/date-time/date-time.module.ts [hunk 2]
```diff
@@ -24,7 +23,7 @@ import { OwlTimerBoxComponent } from './timer-box.component';
 import { OwlTimerComponent } from './timer.component';
 
 @NgModule({
-  imports: [CommonModule, OverlayModule, OwlDialogModule, A11yModule],
```
Comment (pawcoding) [inline projects/picker/src/lib/date-time/date-time.module.ts:27]: 🗑️ 🚀

**Review by pawcoding — APPROVED** [review]
LGTM 👍🏼

File: projects/picker/src/docs/date-time-inline/date-time-inline-demo.component.ts (hunk 1 above)
Comment (NetWin) [reply projects/picker/src/docs/date-time-inline/date-time-inline-demo.component.ts:26]: Hell yeah 💪🏼

**Pull request merged**